src/utils/scopes.js

```diff
@@ -113,6 +122,233 @@ function getThisVariable(frame: any, path: string) {
+    result
+  } = acc;
+  const { type, bindingsNames } = syntheticScope;
+  const key = `${actor}-${scopeIndex + index}`;
+  const isLast = index === lastScopeIndex;
+
+  let title;
+  if (type === "function") {
+    // FIXME Use original function name here
+    const lastGeneratedScope = generatedScopes[generatedScopes.length - 1];
+    const isLastGeneratedScopeFn =
+      lastGeneratedScope && lastGeneratedScope.type === "function";
+    title =
+      isLastGeneratedScopeFn && lastGeneratedScope.function.displayName
+        ? simplifyDisplayName(lastGeneratedScope.function.displayName)
+        : L10N.getStr("anonymous");
+  } else {
+    title = L10N.getStr("scopes.block");
+  }
+
+  let vars = [];
+  bindingsNames.forEach(name => {
+    // Find binding name in the original source bindings
+    const generatedScope = generatedScopes.find(
+      gs => gs.sourceBindings && name in gs.sourceBindings
+    );
+    if (!generatedScope || !generatedScope.sourceBindings) {
+      return;
+    }
+    // .. and map it to the generated name
+    const generatedName = generatedScope.sourceBindings[name];
+    // Skip if we already use the generated name
+    if (generatedName && !foundGeneratedNames[generatedName]) {
+      if (generatedScope.bindings.variables[generatedName]) {
+        vars.push({
+          name,
+          generatedName,
+          path: `${key}/${generatedName}`,
+          contents: generatedScope.bindings.variables[generatedName]
+        });
+        foundGeneratedNames[generatedName] = true;
+        return;
+      }
+
+      const arg = generatedScope.bindings.arguments.find(
+        arg_ => arg_[generatedName]
+      );
+      if (arg) {
+        vars.push({
+          name,
+          generatedName,
+          path: `${key}/${generatedName}`,
+          contents: arg[generatedName]
+        });
+        foundGeneratedNames[generatedName] = true;
+        return;
+      }
+    }
+
+    vars.push({
+      name,
+      generatedName,
+      path: `${key}/${generatedName}`,
+      contents: { value: { type: "undefined" } }
+    });
+  });
+
+  if (isLast) {
+    // For the last synthesized scope, apply all generated names we did not use
+    const allGeneratedVars = generatedScopes.reduce((acc_, { bindings }) => {
+      return acc_.concat(getBindingVariables(bindings, key));
+    }, []);
+    vars = vars.concat(
+      allGeneratedVars.filter(v => !foundGeneratedNames[v.name])
+    );
+  }
+
+  if (index === 0) {
+    // For the first synthesized scope, add this and other vars.
+    if (pauseInfo) {
+      vars = vars.concat(getSpecialVariables(pauseInfo, key));
+    }
+
+    if (selectedFrame) {
+      const this_ = getThisVariable(selectedFrame, key);
+
+      if (this_) {
+        vars.push(this_);
+      }
+    }
+  }
+
+  if (vars && vars.length) {
+    vars.sort((a, b) => a.name.localeCompare(b.name));
+    result.push({
+      name: title,
+      path: key,
+      contents: vars
+    });
+  }
+  return acc;
+}
```

Comment:
It would be nice to create the synthetic scope bindings in the map-bindings space. This would be closer to the data structure we'd get from source-maps

* we can add the special vars in the debugger side